### PR TITLE
Added cmd 0807 Get Tx Power

### PR DESCRIPTION
**Pull request by ISO-B**
Command 0x0807 Get Tx Power doesn't need any parameters. If command is handled successfully response will be first status(0x8000) with success status and after that Get Tx Power Response(0x8807). 0x8807 has only single parameter which is uint8 power. If 0x0807 fails then response is going to be only status(0x8000) with status 1.

Standard power modules of JN516X(Except JN5169) modules have only 4 possible power levels (Table 5 JN-UG-3024 v2.6). These levels are based on some kind of hardware registry so there is no way to change them in firmware. In ZiGate's case this means:

Set/get tx value | Mapped value (dBM)
----------------|----------------------
0 to 31 | 0
32 to 39 | -32
40 to 51 | -20
52 to 63 | -9

JN5169 has 26 levels more information mapping of those can be found from https://community.nxp.com/thread/494272

#145

### CODE REVIEW

**Comment by pipiche38** [comment]
Could you clarify if this is a new implementation of Set TX Power ? comparing to #145 ?
In short does that mean that we have what is describe in #145 for 3.0f and for 3.0a we will have what is described in this PR ?

**Comment by ISO-B** [comment]
I didn't change Set TX Power implementation. We will need to change ZiGate documentation to clarify those mapping ranges. In #145 I just looked first table from documentation that I found and didn't look for more, but while doing this I found other definitions aswell. 

You can still set value between 0x00-0x0bf, but JN516X only cares about lower 6 bits. That means that values greater than 0x3f are mapped against 0x3f. This results always values between 0x00-0x0bf. 

In short there is only 4 (JN516x <- ZiGate) or 26(JN5169) levels that can be used. Handling set value to level happens in hardware.

**Comment by doudz** [comment]
Maybe the response 8807 could be on Set TX too , in order to know the real TX power

**Comment by ISO-B** [comment]
So it would return mapped value(0, -9, -20, -32) instead of 0-63?  Is there any case where it would be desired to set lower power than full power?

**Comment by pipiche38** [comment]
In such , shall we not have the SetTXPower having the same approach: 4 levels from 1 to 4 . 
4 Max and 1 Low.

You might want to reduce the power as you do on the Wifi, just to reduce to wave exposure , no ?

Maybe stupid question, but I agree somehow with you, what was the purpose of introducing it in 3.0f .

**Comment by ISO-B** [comment]
> You might want to reduce the power as you do on the Wifi, just to reduce to wave exposure , no ?
Yes, I didn't think that since I have no neighbors near by.

I don't know is someone using JN5169 with ZiGate firmware or do we need to take that into account. If we need to think that then we would need make multiple parsers for different chips. There is also two JN5168 high power models which both have different four levels.

What about if we return both uint value and JN5168 mapped power level? Then it would serve other chips too even if mapped power level is incorrect for those.

**Comment by pipiche38** [comment]
I'll like your proposal. This is quiet open and should indeed satisfy everybody

**Comment by ISO-B** [comment]
Now Set Tx command will respond also with set power level. This is now ready for testing and merge.

**Comment by fairecasoimeme** [comment]
Ok, It's good for me, It's works !

**Comment by pipiche38** [comment]
confirmed on 3.1a
